src/Atc.Rest.MinimalApi/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,35 @@
+namespace Atc.Rest.MinimalApi.Extensions;
+
+internal static class DictionaryExtensions
+{
+    /// <summary>
+    /// Merges two dictionaries containing errors and ensures there are no duplicate values for the same key.
+    /// </summary>
+    /// <param name="errorsA">The first dictionary of errors to merge.</param>
+    /// <param name="errorsB">The second dictionary of errors to merge.</param>
+    /// <returns>A new dictionary which includes the keys and values from both input dictionaries.
+    /// If the same key exists in both dictionaries, the values are merged and duplicates are removed.</returns>
+    internal static IDictionary<string, string[]> MergeErrors(
+        this IDictionary<string, string[]> errorsA,
+        IDictionary<string, string[]> errorsB)
+    {
+        var result = new Dictionary<string, string[]>(StringComparer.Ordinal);
```

Comment:
Sorry I didn't catch this before. There's no reason to create a third dictionary here, outside the if statement..
Either just `return new Dictionary<string, string[]>(StringComparer.Ordinal);` or return `errorsA` or `errorsB` and avoid instantiating a new object all together.

Reply:
Returning a new dictionary instead of creating it as a variable first would be a good idea. You shouldn't return one of the input dictionary as it would be misleading the consumer of this method to return the same instance as what was passed in

Reply:
But it's already returning the `errorsA` instance, if there's any errors ?

Reply:
Fixed